enumerable.rb

```diff
@@ -0,0 +1,153 @@
+# rubocop:disable all
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    is_a?(Array) ? self : to_a
+    temp = []
+    my_each do |j|
+      yield(j) ? temp.push(j) : next
+    end
+    temp
+  end
+
+  def my_all?(*arg)
+
+    return grep(arg.first).length == size unless arg.empty?
+
+    my_each { |el| return false unless yield(el) } if block_given?
+    my_each { |el| return false unless el } unless block_given?
+    true
+  end
+
+  def my_any?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    current = self
+    result = false
+    if !block_given? && pattern.nil?
+      current.my_each do |item|
+        result = true if item
+      end
+    elsif pattern.is_a? Regexp
+      current.my_each do |item|
+        result = true if item =~ pattern
+      end
+    elsif pattern.is_a? Class
+      current.my_each do |item|
+        result = true if item.is_a? pattern
+      end
+    elsif pattern
+      current.my_each do |item|
+        result = true if item == pattern
+      end
+    elsif is_a? Array
+      current.my_each do |item|
+        result = true if yield(item)
+      end
+    elsif is_a? Hash
+      current.my_each do |k, v|
+        result = true if yield(k, v)
+      end
+    end
+    result
+  end
+
+  def my_none?(*arg)
+    return grep(arg.first).empty? unless arg.empty?
+
+    my_each { |el| return false if yield(el) } if block_given?
+    my_each { |el| return false if el } unless block_given?
+    true
+  end
+
+  def my_count(parameter = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(i) }
+    elsif !parameter.nil?
+      my_each { |i| count += 1 if i == parameter }
+    else
+      count = length
+    end
+    count
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    i = 0
+    new_map = []
+    while i < size
+      new_map << yield(self[i])
+      i += 1
+    end
+    new_map
+  end
+
+  def my_inject(pattern = nil, str = nil)
+    tmp = is_a?(Range) ? to_a : self
+    memo = pattern.nil? || pattern.is_a?(Symbol) ? tmp[0] : pattern
+    if block_given?
+      start = pattern ? 0 : 1
+      tmp[start..-1].my_each { |e| memo = yield(memo, e) }
+    end
+    tmp[1..-1].my_each { |e| memo = memo.send(pattern, e) } if acc.is_a?(Symbol)
+    tmp[0..-1].my_each { |e| memo = memo.send(str, e) } if cur
+    memo
+  end
```

Comment:
There are some inconsistency, replace **acc** on line 116 to **pattern** and also **cur** to **str** on line 117.

Reply:
ooohh i will fix that